plugins/modules/proxmox.py

```diff
@@ -1067,257 +696,928 @@ def main():
+            time.sleep(1)
 
-            delete_params = {}
+        self.module.fail_json(
+            vmid=vmid,
+            taskid=taskid,
+            msg="%sLast line in task before timeout: %s" % (timeout_msg, self.proxmox_api.nodes(node).tasks(taskid).log.get()[:1])
+        )
 
-            if module.params['purge']:
-                delete_params['purge'] = 1
+    def is_template_container(self, node: str, target: int) -> bool:
+        """Check if the specified container is a template."""
+        proxmox_node = self.proxmox_api.nodes(node)
+        config = getattr(proxmox_node, self.VZ_TYPE)(target).config.get()
+        return config.get('template', False)
 
-            taskid = getattr(proxmox.proxmox_api.nodes(vm['node']), VZ_TYPE).delete(vmid, **delete_params)
+    def content_check(self, node: str, ostemplate: str, template_store: str) -> bool:
+        """Check if the specified ostemplate is present in the specified storage."""
+        proxmox_node = self.proxmox_api.nodes(node)
+        storage_contents = proxmox_node.storage(template_store).content.get()
+        return any(content['volid'] == ostemplate for content in storage_contents)
+
+    def validate_tags(self, tags: List[str]) -> bool:
+        """Check if the specified tags are valid."""
+        re_tag = re.compile(r'^[a-z0-9_][a-z0-9_\-\+\.]*$')
+        for tag in tags:
+            if not re_tag.match(tag):
+                self.module.fail_json(msg="%s is not a valid tag" % tag)
+                return False
+        return True
 
-            while timeout:
-                if proxmox.api_task_ok(vm['node'], taskid):
-                    module.exit_json(changed=True, vmid=vmid, taskid=taskid, msg="VM %s removed" % vmid)
-                timeout -= 1
-                if timeout == 0:
-                    module.fail_json(vmid=vmid, taskid=taskid, msg='Reached timeout while waiting for removing VM. Last line in task before timeout: %s'
-                                     % proxmox.proxmox_api.nodes(vm['node']).tasks(taskid).log.get()[:1])
+    def check_supported_features(self) -> None:
+        for option, version in self.MINIMUM_VERSIONS.items():
+            if self.version() < LooseVersion(version) and option in self.module.params:
+                self.module.fail_json(
+                    changed=False,
+                    msg="Feature {option} is only supported in PVE {version}+, and you're using PVE {pve_version}".format(
+                        option=option, version=version, pve_version=self.version()
+                    ),
+                )
 
-                time.sleep(1)
-        except Exception as e:
-            module.fail_json(vmid=vmid, msg="deletion of VM %s failed with exception: %s" % (vmid, to_native(e)))
+
+def main():
+    module = get_ansible_module()
+    proxmox = ProxmoxLxcAnsible(module)
+
+    try:
+        proxmox.run()
+    except Exception as e:
+        module.fail_json(msg="An error occurred: %s" % to_native(e))
```

Comment:
You might want to consider using `ModuleHelper`. See more in https://docs.ansible.com/ansible/latest/collections/community/general/docsite/guide_modulehelper.html

Particularly for this module, I would recommend using `StateModuleHelper`, as it is already tailored for modules with different behaviour for each `state`. 

There is a number of modules already using these helpers, you can spot them using 
```bash
git grep -E '\(\S*ModuleHelper\)'
```

Reply:
see [this comment](https://github.com/ansible-collections/community.general/pull/9225#issuecomment-2548649397) for a reply